### PR TITLE
Adding sawtooth waveform example to repository

### PR DESCRIPTION
**Pull request by miketaschuk**
Brief; This pull request adds an example script that produces a sawtooth waveform on the output of the pico. 

Files added

- ./examples/example_sawtooth.py; contains a SawtoothWaveform class and  a script to run a demo of the class
- ./test/test_sawtooth; contains a MockPico class for testing, and tests for SawtoothWaveform

Testing Protocol
- extensive unit tests for the new sawtooth class are included in this commit; the code is passing all of them
- extensive physical tests on a pico, with report summarizing the testing [here](https://docs.google.com/document/d/1SiSRDVOIwgOKPM2i2j9lLm8cGcY0gzg-rcGwRhnfvtM/edit)

Issues Addressed:

- Issue #7
- Issue #10 
- Issue #11 
- Issue #12

### CODE REVIEW

File: test/test_sawtooth.py [hunk 1]
```diff
@@ -0,0 +1,425 @@
+#!/usr/bin/env python3
+
+import datetime as dt
+import unittest
+import sys
+import warnings
+
+from examples.example_sawtooth import SawtoothWaveform
+from examples.example_sawtooth import TimestepWarning
```
Comment (chris-g2v) [inline test/test_sawtooth.py:9]: These two lines throw an error saying that the example module can not be found.

**Review by chris-g2v — CHANGES_REQUESTED** [review]
The existing code base tries its best to adhere to the PEP 8 Python style guide. Please update the two sawtooth files to use spaces instead of tabs to keep with the consistency.

File: test/test_sawtooth.py (hunk 1 above)
Comment (miketaschuk) [reply test/test_sawtooth.py:9]: Please share how you are running the tests. The Makefile is working currently, so I think the difference is in how the code is being called.

Comment (chris-g2v) [reply test/test_sawtooth.py:9]: Yes, that was the problem. Calling the file inside the _test_ directory causes this error to occur.

**Review by chris-g2v — APPROVED** [review]

**Pull request merged**